skipchain/struct_test.go

```diff
@@ -118,11 +118,11 @@ func TestSkipBlock_Hash2(t *testing.T) {
 
 func TestBlockLink_Copy(t *testing.T) {
 	// Test if copy is deep or only shallow
-	b1 := &BlockLink{}
-	b1.Sig = []byte{1}
+	b1 := &ForwardLink{}
+	b1.Signature.Sig = []byte{1}
 	b2 := b1.Copy()
-	b2.Sig = []byte{2}
-	if bytes.Equal(b1.Sig, b2.Sig) {
+	b2.Signature.Sig = []byte{2}
```

Comment:
This test doesn't make sense, it's testing something obvious (set to two different slices, then check that they are different). It would make more sense to mutate b1.Signature after the Copy() and confirm that b2 does NOT see the change made to b1. i.e. b1.Signature.Sig[0] = 0.